### ISSUE TITLE
Jobs stuck in waiting state

### ISSUE THREAD

**Issue by spficklin** [bcuser30/ML_GalaxyQueue#9]
It seems jobs on the test server get stuck in the waiting state. As reported by MingChen0919  here:  tripal/tripal_galaxy/issues/101.

I was able to duplicate this problem by submitting a job. It sits in the queue waiting after submission through the Tripal Galaxy Module.  If I then go to the Galaxy web portal for the test server and go to my history I can manually click the icon to re-run the job and then it will start. 

So, I think of two reasons why this would happen.  
1.  There's an updated to the Galaxy API that requires we set a flag when invoking the workflow to get it going
2.  There's some configuration setting somewhere that controls this behavior.

Are you aware of anything on that second item?

**Comment by heh30** [comment]
There is not an obvious configuration setting in the galaxy.yml file that would cause the different between test and production.  I'll keep digging, it's possible something new/different was introduced with the newer version that causes this problem.  

Regarding the first item, have you adjusted the flag and tested?

**Comment by spficklin** [comment]
Oh, I don't know if there is a flag or not, it was just the only two ideas I could come up with for why those jobs are hanging.  I will look into it.

Is there anything in the Galaxy logs related to job submission waiting?

**Comment by heh30** [comment]
I do know the ML cluster is seeing much more activity now that nearly everyone has returned from the conference.  This may or may not be related to the differences between prod and test.  Remember, prod is running jobs locally, not via cluster.

I’ve enabled some additional logging, please replicate the problem.

Regards,
Heidi

**Comment by MingChen0919** [comment]
Hi @bcuser30 , I submitted a testing job this morning. The job didn't even run (history window is gray). After a while, I got this error:

 
<img width="853" alt="screen shot 2018-06-08 at 11 45 48 am" src="https://user-images.githubusercontent.com/1262709/41167845-1b873ad4-6b12-11e8-9a42-224dd4a41762.png">

**Comment by heh30** [comment]
Please try again

**Comment by MingChen0919** [comment]
tried. i got the error "The cluster DRM system terminated this job" in the middle of the job.

<img width="423" alt="screen shot 2018-06-08 at 12 23 51 pm" src="https://user-images.githubusercontent.com/1262709/41169431-ff43beb0-6b16-11e8-838d-c2f6e23f788a.png">

**Comment by heh30** [comment]
Progress.  What that tells me is that there's a problem starting galaxy in daemon mode and having it recognize the cluster at all.  

The job partially completed then?  Later portions of the workflow got the DRM error?  Does it always break on this step or is it random?

**Comment by MingChen0919** [comment]
Yes, the job partially completed. I think it's random. Let me try a new submission and see where it breaks.

**Comment by MingChen0919** [comment]
this time it completed successfully! so the DRM error is random.

**Comment by heh30** [comment]
That's horribly aggravating.  I'd adjusted what I thought was a timeout value yesterday.  Our cluster is not nearly as busy today as it was yesterday (so far).  I'll keep trying to identify if there are other timeout settings that can be configured.

**Comment by heh30** [comment]
Log details regarding the failure

galaxy.jobs.runners.drmaa DEBUG 2018-06-08 09:23:26,255 [p:3612,w:1,m:0] [Dummy-5] (4223/3234229) state change: job is running
galaxy.jobs.runners.drmaa DEBUG 2018-06-08 09:23:28,453 [p:3612,w:1,m:0] [Dummy-5] (4223/3234229) state change: job finished, but failed
galaxy.jobs DEBUG 2018-06-08 09:23:28,557 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] fail(): Moved /galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_dataset_6290.dat to /galaxy/
test/htdocs/database/files/006/dataset_6083.dat
galaxy.jobs DEBUG 2018-06-08 09:23:28,557 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] fail(): Moved /galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_dataset_6291.dat to /galaxy/
test/htdocs/database/files/006/dataset_6084.dat
galaxy.tools.error_reports DEBUG 2018-06-08 09:23:28,853 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] Bug report plugin <galaxy.tools.error_reports.plugins.sentry.SentryPlugin object at 0x7
ff90cf0de10> generated response None
galaxy.tools.error_reports DEBUG 2018-06-08 09:23:28,853 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] Bug report plugin <galaxy.tools.error_reports.plugins.sentry.SentryPlugin object at 0x7
ff90cf0de10> generated response None
galaxy.model.metadata DEBUG 2018-06-08 09:23:28,857 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] Cleaning up external metadata files
galaxy.model.metadata DEBUG 2018-06-08 09:23:28,890 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] Failed to cleanup MetadataTempFile temp files from /galaxy/test/htdocs/database/jobs_directo
ry/004/4223/metadata_out_HistoryDatasetAssociation_6290_Vgl6qR: No JSON object could be decoded
galaxy.model.metadata DEBUG 2018-06-08 09:23:28,890 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] Failed to cleanup MetadataTempFile temp files from /galaxy/test/htdocs/database/jobs_directo
ry/004/4223/metadata_out_HistoryDatasetAssociation_6291_ueBXRl: No JSON object could be decoded
galaxy.jobs.runners DEBUG 2018-06-08 09:23:28,930 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] (4223/3234229) Unable to cleanup /galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4
223.sh: [Errno 2] No such file or directory: '/galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4223.sh'
galaxy.jobs.runners DEBUG 2018-06-08 09:23:28,945 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] (4223/3234229) Unable to cleanup /galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4
223.o: [Errno 2] No such file or directory: '/galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4223.o'
galaxy.jobs.runners DEBUG 2018-06-08 09:23:28,962 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] (4223/3234229) Unable to cleanup /galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4
223.e: [Errno 2] No such file or directory: '/galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4223.e'
galaxy.jobs.runners DEBUG 2018-06-08 09:23:28,977 [p:3612,w:1,m:0] [DRMAARunner.work_thread-0] (4223/3234229) Unable to cleanup /galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4
223.ec: [Errno 2] No such file or directory: '/galaxy/test/htdocs/database/jobs_directory/004/4223/galaxy_4223.ec'

**Comment by heh30** [comment]
Error reported by cluster job

qacct -j 3234229
==============================================================
qname        web.q
hostname     node13
group        mainwadm
owner        galaxy
project      NONE
department   defaultdepartment
jobname      g4223_fastqc_ming_chen0919_gmail_com
jobnumber    3234229
taskid       undefined
account      sge
priority     10
qsub_time    Fri Jun  8 09:23:12 2018
start_time   -/-
end_time     -/-
granted_pe   NONE
slots        1
failed       26  : opening input/output file
exit_status  0
ru_wallclock 0
ru_utime     0.000
ru_stime     0.000
ru_maxrss    0
ru_ixrss     0
ru_ismrss    0
ru_idrss     0
ru_isrss     0
ru_minflt    0
ru_majflt    0
ru_nswap     0
ru_inblock   0
ru_oublock   0
ru_msgsnd    0
ru_msgrcv    0
ru_nsignals  0
ru_nvcsw     0
ru_nivcsw    0
cpu          0.000
mem          0.000
io           0.000
iow          0.000
maxvmem      0.000
arid         undefined

**Comment by spficklin** [comment]
So the error log doesn't seem very helpful. It's not clear if the messages after the "Job finished, but failed" are the cause of the error or resulting from the job failing.  

I don't know enough about how Galaxy operates to figure that one out.  Do you have any ideas?

**Comment by heh30** [comment]
I also don't understand the details of the inner-workings.  What I suspect is that the jobs_directory for that particular job is being removed before all the jobs are completed.  I'm not sure how to prove that, since this problem is so random.

**Comment by MingChen0919** [comment]
@bcuser30 , you are right. For every job, galaxy creates a temporary working directory. Once the job is complete, whether successfully or not, that temporary working directory is removed.

**Comment by heh30** [comment]
Do the workflows all use the same temporary working directory?

**Comment by MingChen0919** [comment]
No. each job has its own temporary working directory. But they are all under the directory `/main/sites/galaxy/test-galaxy/htdocs/database/jobs_directory`.

See those folders named with numbers below. Each of those folders is a running job. Once the job is completed. These folders will be deleted automatically.

<img width="637" alt="screen shot 2018-06-08 at 4 13 20 pm" src="https://user-images.githubusercontent.com/1262709/41178930-e734f95e-6b36-11e8-8bf2-3b98da202183.png">

**Comment by heh30** [comment]
The qacct output indicates the job failed '26 : opening input/output file' 

That's still pretty cryptic, but tells me the cluster had problems accessing the file.  Why....I'm not sure.

**Comment by MingChen0919** [comment]
Not sure if this information is helpful:

Many jobs have status "new" and they failed. They don't have a working folder in the job directory.

<img width="1145" alt="screen shot 2018-06-08 at 4 20 29 pm" src="https://user-images.githubusercontent.com/1262709/41179293-42970a66-6b38-11e8-8f69-a45a5a626ab1.png">

**Comment by heh30** [comment]
Jobs within  the workflow depend on outputs of previous jobs?  Is the workflow set up in such a way that it waits for jobs to finish before starting a new step?

The only galaxy job currently running on our cluster is this g4198_ncbi_blastp_wrapper_stephen_ficklin_wsu_edu

I think we're starting to talk about both tickets together; the DRM issue and also the wait issue.  The DRM issue seems to be related to the directory being removed at the wrong time / before the job is complete.  

The wait issue, I couldn't find settings for and have had difficulty tracking down anything of value in the log.

**Comment by MingChen0919** [comment]
Yes, Jobs within the workflow depend on outputs of previous jobs. 

You know what, I am going to submit 10 workflows and see how many of them will complete successfully. If the DRM issue is fixed. This issue may be gone.

**Comment by MingChen0919** [comment]
I finished 5 workflows, all 5 have an DRM issue somewhere in the workflow. I'll comment on the DRM issue thread.

**Comment by spficklin** [comment]
Just to clarify what @MingChen0919 said: " you are right. For every job, galaxy creates a temporary working directory. "

It's my understanding that galaxy creates these temp directories for the workflow.  We don't control that with our job submissions. We just submit the workflow.
